diff --git a/gemrb/core/Scriptable/Actor.cpp b/gemrb/core/Scriptable/Actor.cpp
index 1ec11d4b..6d7ffa60 100644
--- a/gemrb/core/Scriptable/Actor.cpp
+++ b/gemrb/core/Scriptable/Actor.cpp
@@ -7861,7 +7861,7 @@ void Actor::UpdateActorState()
 		return;
 	}
 
-	Animation* first = anim[0].first;
+	Animation* first = anim[0].first.get();
 
 	if (first->endReached) {
 		// possible stance change
@@ -7870,7 +7870,7 @@ void Actor::UpdateActorState()
 			first->endReached = false;
 			first->SetFrame(0);
 
-			Animation* firstShadow = currentStance.shadow.empty() ? nullptr : currentStance.shadow[0].first;
+			Animation* firstShadow = currentStance.shadow.empty() ? nullptr : currentStance.shadow[0].first.get();
 			if (firstShadow) {
 				firstShadow->endReached = false;
 				firstShadow->SetFrame(0);
@@ -8253,7 +8253,7 @@ void Actor::DrawActorSprite(const Point& p, BlitFlags flags,
 	}
 
 	for (const auto& part : animParts) {
-		const Animation* anim = part.first;
+		const Animation* anim = part.first.get();
 		Holder<Palette> palette = part.second;
 
 		Holder<Sprite2D> currentFrame = anim->CurrentFrame();
@@ -8343,21 +8343,21 @@ bool Actor::AdvanceAnimations()
 	for (int part = 0; part < count; ++part) {
 		int partnum = part;
 		if (zOrder) partnum = zOrder[part];
-		Animation* anim = stanceAnim->at(partnum).get();
+		auto anim = stanceAnim->at(partnum);
 		if (anim) {
 			currentStance.anim.emplace_back(anim, anims->GetPartPalette(partnum));
 		}
 
 		if (shadows) {
-			Animation* shadowAnim = shadows->at(partnum).get();
+			auto shadowAnim = shadows->at(partnum);
 			if (shadowAnim) {
 				currentStance.shadow.emplace_back(shadowAnim, anims->GetShadowPalette());
 			}
 		}
 	}
 
-	Animation* first = currentStance.anim[0].first;
-	Animation* firstShadow = currentStance.shadow.empty() ? nullptr : currentStance.shadow[0].first;
+	Animation* first = currentStance.anim[0].first.get();
+	Animation* firstShadow = currentStance.shadow.empty() ? nullptr : currentStance.shadow[0].first.get();
 
 	// advance first (main) animation by one frame (in sync)
 	if (Immobile()) {
@@ -8529,7 +8529,7 @@ void Actor::UpdateDrawingRegion()
 
 	auto ExpandBoxForAnimationParts = [&box, this](const std::vector<AnimationPart>& parts) {
 		for (const auto& part : parts) {
-			const Animation* anim = part.first;
+			const Animation* anim = part.first.get();
 			Holder<Sprite2D> animframe = anim->CurrentFrame();
 			if (!animframe) continue;
 			Region partBBox = animframe->Frame;
diff --git a/gemrb/core/Scriptable/Actor.h b/gemrb/core/Scriptable/Actor.h
index ab33f8e3..a1c91967 100644
--- a/gemrb/core/Scriptable/Actor.h
+++ b/gemrb/core/Scriptable/Actor.h
@@ -524,7 +524,7 @@ private:
 
 	CharAnimations* anims = nullptr;
 
-	using AnimationPart = std::pair<Animation*, Holder<Palette>>;
+	using AnimationPart = std::pair<Holder<Animation>, Holder<Palette>>;
 	struct {
 		std::vector<AnimationPart> anim;
 		std::vector<AnimationPart> shadow;
