diff --git a/gemrb/core/TraversabilityCache.h b/gemrb/core/TraversabilityCache.h
--- a/gemrb/core/TraversabilityCache.h
+++ b/gemrb/core/TraversabilityCache.h
@@ -110,6 +110,11 @@
 		return traversabilityData.size();
 	}
 
+	void Initialize()
+	{
+		ValidateTraversabilityCacheSize();
+	}
+
 	void Update();
 
 private:
diff --git a/gemrb/core/Map.cpp b/gemrb/core/Map.cpp
--- a/gemrb/core/Map.cpp
+++ b/gemrb/core/Map.cpp
@@ -441,6 +441,7 @@
 {
 	area = this;
 	MasterArea = core->GetGame()->MasterArea(scriptName);
+	traversabilityCache.Initialize();
 }
 
 Map::~Map(void)
